### ISSUE TITLE
no-script-url fires if `javascript:` is found anywhere in the code

### ISSUE THREAD

**Issue by bluesmoon** [eslint/eslint#1448]
no-script-url should fire only if a `javascript:` url is assigned to `location.href` or to some place where it can be executed.

For example, we have code that checks if a resource's url starts with `javascript:` so that we can ignore those urls.  The code looks like this:

``` javascript
if(e.name.indexOf('about:') === 0 ||
    e.name.indexOf('javascript:') === 0) {
        continue;
}
```

And it gets flagged by the rule.
